### PR TITLE
Mem bio leak fix2

### PR DESCRIPTION
**Pull request by cminyard**
##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/bio_memleak_test.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    bufmem.length = strlen(str) + 1;
+    bufmem.data = (char *) str;
+    bufmem.max = bufmem.length;
+    BIO_set_mem_buf(bio, &bufmem, BIO_NOCLOSE);
+    BIO_set_flags(bio, BIO_FLAGS_MEM_RDONLY);
+
+    if (BIO_read(bio, buf, sizeof(buf)) <= 0)
+	goto finish;
+
+    if (strcmp(buf, str) != 0)
```
Comment (FdaSilvaYY) [inline test/bio_memleak_test.c]: style: *IMHO, I will write this more shortly:
`     ok = strcmp(buf, str) == 0;`

Comment (cminyard) [reply test/bio_memleak_test.c]: Is this official openssl style?  I personally think that what you suggest is harder to read, but I know opinions on this vary.  I'm not sure who is official and who makes the decisions, but I can change this if necessary.

Comment (cminyard) [reply test/bio_memleak_test.c]: I guess to be more clear, what the way I wrote it is more consistent with the rest of the function, and that's why I preferred that way.  Though "finish" might be better named "err" or something like that.

Comment (mattcaswell) [reply test/bio_memleak_test.c]: Either form would be acceptable according to OpenSSL style

**Comment by cminyard** [comment]
A lot of stuff changed in openssl since I put up the request, some renames are probably getting you.  Though I'm not sure why.  Anyway, I have a new branch, I've tested, and I've changed the style thing with strcmp.  So this should be all goodness.

**Comment by cminyard** [comment]
I created pull request 8051 to replace this.

**Pull request closed without merging**